sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedParquetRecordReader.java

```diff
@@ -197,11 +200,17 @@ public void initBatch(MemoryMode memMode, StructType partitionColumns,
   }
 
   public void initBatch() {
-    initBatch(DEFAULT_MEMORY_MODE, null, null);
+    assert(memoryMode != null);
+    initBatch(memoryMode,null, null);
```

Comment:
[Here ](https://github.com/apache/spark/pull/17436/files/e8c598097573ff469eddf2b2a8f273b6ea33fc86#diff-ee26d4c4be21e92e92a02e9f16dbc285R359) is a place to pass the 3rd argument.